server/database.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @fileoverview database function handler
+ */
+var assert = require('assert');
+
+var log = require('./log.js');
+
+/** @type {database} */
+module.exports = database;
+
+/**
+ * @constructor
+ */
+function database() {}
+
+/**
+ * Checks the user information for signing in.
+ * @param {!Object} collection Database collection.
+ * @param {!Object} query Database query.
```

Comment:
But it seems that the database query you passed in are simply objects, rather than any compound objects.
```js
/**
 * @typedef {{
 *    $and: (!Object|undefined),
 *    $or: (!Object|undefined)
 *  }}
 */
mongdb.Query;
```